src/ovp_pipeline/knowledge_index.py

```
@@ -1010,6 +1107,116 @@ def search_knowledge_index(vault_dir: Path, query: str, limit: int = 10) -> list
+        from datetime import datetime, timezone
+
+        now_ts = int(datetime.now(timezone.utc).timestamp())
+    tau_seconds = max(tau_days * 86400.0, 1.0)
+
+    metrics: dict[str, tuple[int, int, int]] = {}
+    titles_db: dict[str, str] = {}
+    with sqlite3.connect(layout.knowledge_db) as conn:
+        placeholders = ",".join("?" for _ in rrf_scores)
+        for slug, last_seen_ts, reuse_count, citation_count in conn.execute(
+            f"SELECT slug, last_seen_ts, reuse_count, citation_count "
+            f"FROM page_metrics WHERE slug IN ({placeholders})",
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)